#163602556 persist data in database

Jesse-efe opened the pull request:
#### What does this PR do?
Integrate postgres database to project
#### Description of Task to be completed?

Persist App data in database

Write SQL queries to store and get data from database

Test API endpoints
#### How should this be manually tested?
- Pull this repository and cd into it
- Run npm install to install all dependencies
- Run npm start to start the app
- Use Postman to test the routes listed readme.md

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#163552796 #163602556
#### Screenshots
N/A
#### Questions:
N/A

Jesse-efe committed:
feat(server/test): persist data in database

- integrate postgres database to project
- store and retrieve data from database
- modify test

[delivers #163602556]

Jesse-efe committed:
chore(test): add postgres service in travis.yml file

[finishes #163612492]

akinwale-habibullah reviewed (commented):
Add a sample.env file in your project root folder, and add the required variables that your project needs to successfully connect to the db.

Another thing I will like to see is a guide in your README file, detailing what I need to configure to get your app running locally.

@Jesse-efe

Jesse-efe commented:
Ok

Jesse-efe committed:
chore(server): create script to create tables automatically

-Update readme.md with installation guide to use
-Installation guide should include steps on how to create tables

[finishes #163619453]

akinwale-habibullah reviewed (commented):
You dont have a sample.env file in your repo as you instructed in your readme. @Jesse-efe

akinwale-habibullah commented on source/models/config.js:
```diff
@@ -0,0 +1,10 @@
+import { Pool } from 'pg';
+
+require('dotenv').config();
```
Why using require along with import ?

akinwale-habibullah commented on source/models/tables.js:
```diff
@@ -0,0 +1,43 @@
+  console.log('created');
+};
+
+export default seedDb;
```
Please add some data to your app. and create at least one admin record, which uses an environment variable specified by the user.

akinwale-habibullah commented on source/models/config.js:
```diff
@@ -0,0 +1,10 @@
+
+const pool = new Pool({
+  // connectionString: process.env.devConnectionString,
+  connectionString: process.env.testConnectionString,
```
You need to have better logic to help automate this, instead of relying on the user to change values here.

Jesse-efe committed:
chore(server): seed dabase on installation

[finishes  #163625461]

Jesse-efe commented:
Ok thanks, I have implemented the feedback

Jesse-efe committed:
fix(bug fix): fix bug with seeding database

- phoneNumber should be an INT not a string

[fixes #163627496]
